src/braket/circuits/text_diagram_builders/unicode_circuit_diagram.py

```diff
@@ -0,0 +1,282 @@
+            isinstance(item, Instruction) and isinstance(item.operator, CompilerDirective)
+        ):
+            target_qubits = circuit_qubits
+            control_qubits = QubitSet()
+            qubits = circuit_qubits
+            ascii_symbols = [item.ascii_symbols[0]] * len(qubits)
+            UnicodeCircuitDiagram._update_connections(qubits, connections)
+        elif (
+            isinstance(item, Instruction)
+            and isinstance(item.operator, Gate)
+            and item.operator.name == "GPhase"
+        ):
+            target_qubits = circuit_qubits
+            control_qubits = QubitSet()
+            qubits = circuit_qubits
+            ascii_symbols = UnicodeCircuitDiagram._qubit_line_character() * len(circuit_qubits)
+        else:
+            if isinstance(item.target, list):
+                target_qubits = reduce(QubitSet.union, map(QubitSet, item.target), QubitSet())
+            else:
+                target_qubits = item.target
+            control_qubits = getattr(item, "control", QubitSet())
+            control_state = getattr(item, "control_state", "1" * len(control_qubits))
+            map_control_qubit_states = {
+                qubit: state for qubit, state in zip(control_qubits, control_state)
+            }
+
+            target_and_control = target_qubits.union(control_qubits)
+            qubits = QubitSet(range(min(target_and_control), max(target_and_control) + 1))
+            ascii_symbols = item.ascii_symbols
+            UnicodeCircuitDiagram._update_connections(qubits, connections)
+
+        return (
+            target_qubits,
+            control_qubits,
+            qubits,
+            connections,
+            ascii_symbols,
+            map_control_qubit_states,
+        )
+
+    @staticmethod
+    def _update_connections(qubits: QubitSet, connections: dict[Qubit, str]) -> None:
+        if len(qubits) > 1:
+            connections |= {qubit: "both" for qubit in qubits[1:-1]}
+            connections[qubits[-1]] = "above"
+            connections[qubits[0]] = "below"
+
+    # Ignore flake8 issue caused by Literal["above", "below", "both", "none"]
+    # flake8: noqa: BCS005
+    @classmethod
+    def _draw_symbol(
+        cls,
+        symbol: str,
+        symbols_width: int,
+        connection: Literal["above", "below", "both", "none"],
+    ) -> str:
+        """Create a string representing the symbol inside a box.
+
+        Args:
+            symbol (str): the gate name
+            symbols_width (int): size of the expected output. The ouput will be filled with
+                cls._qubit_line_character() if needed.
+            connection (Literal["above", "below", "both", "none"]): specifies if a connection
+                will be drawn above and/or below the box.
+
+        Returns:
+            str: a string representing the symbol.
+        """
+        top = ""
+        bottom = ""
+        if symbol in ["C", "N", "SWAP"]:
```

Comment:
I'm not crazy about hard-coding these (again?)... couldn't we just have put in the correct symbol on line 131? For SWAP, why are we not just changing the symbol in the class, rather than here?